### PR TITLE
AddBox endpoint added to add box to scans

### LINKED ISSUE #1112: Implement addBox from EIP-1
Add "addBox" API method described in EIP-1 to associate a box with an application.

### PR DESCRIPTION
**Pull request by kushti**
Close #1112 : this PR is introducing /scan/addBox API endpoint to add a box to specific scans, or change scans of boxes already stored in the wallet database.

### CODE REVIEW

File: src/main/resources/api/openapi.yaml [hunk 1]
```diff
@@ -3574,6 +3587,35 @@ paths:
...
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ApiError'
+
+  /scan/addBox:
+    post:
+      security:
+        - ApiKeyAuth: [api_key]
+      summary: Add scan to a box, writes box to database if it is not there. You can use scan number 10 to add a box to the wallet.
```
Comment (aslesarenko) [inline src/main/resources/api/openapi.yaml:3601]: Maybe "add the box to the scans"?

File: src/test/scala/org/ergoplatform/nodeView/wallet/persistence/WalletRegistrySpec.scala [hunk 2]
```diff
@@ -21,7 +21,7 @@ class WalletRegistrySpec
     with WalletGenerators
     with FileUtils {
 
-  implicit override val generatorDrivenConfig = PropertyCheckConfiguration(minSuccessful = 5, sizeRange = 10)
+  implicit override val generatorDrivenConfig = PropertyCheckConfiguration(minSuccessful = 4, sizeRange = 10)
```
Comment (aslesarenko) [inline src/test/scala/org/ergoplatform/nodeView/wallet/persistence/WalletRegistrySpec.scala:24]: It is too small for property testing. What is the reason to decrease it?

File: src/test/scala/org/ergoplatform/nodeView/wallet/persistence/WalletRegistrySpec.scala [hunk 3]
```diff
@@ -204,6 +204,27 @@ class WalletRegistrySpec
...
+  it should "update scans correctly" in {
+    val appId1: ScanId = ScanId @@ 21.toShort
+    val appId2: ScanId = ScanId @@ 22.toShort
+
+    forAll(trackedBoxGen) { tb0 =>
+      withHybridStore(10) { store =>
+        val tb1 = tb0.copy(scans = Set(appId1, appId2), spendingHeightOpt = None, spendingTxIdOpt = None)
+
+        val reg = new WalletRegistry(store)(ws)
+        WalletRegistry.putBox(emptyBag, tb1).transact(store)
```
Comment (aslesarenko) [inline src/test/scala/org/ergoplatform/nodeView/wallet/persistence/WalletRegistrySpec.scala:216]: Looks like a backdoor access to store. Why not use registry to put the box?

File: src/test/scala/org/ergoplatform/nodeView/wallet/persistence/WalletRegistrySpec.scala (hunk 2 above)
Comment (kushti) [reply src/test/scala/org/ergoplatform/nodeView/wallet/persistence/WalletRegistrySpec.scala:24]: With default value test is going to be slow and breaking Travis. The reason here is purely heuristic, more tests we have, less runs for a test we can have.

File: src/main/resources/api/openapi.yaml [hunk 4]
```diff
@@ -850,6 +850,19 @@ components:
         boxId:
           $ref: '#/components/schemas/TransactionBoxId'
 
+    ScanIdsBox:
+      type: object
```
[the remark this answers is not in the document]
Comment (kushti) [reply src/main/resources/api/openapi.yaml:854]: done

File: src/main/resources/api/openapi.yaml (hunk 1 above)
Comment (kushti) [reply src/main/resources/api/openapi.yaml:3619]: done

File: src/test/scala/org/ergoplatform/nodeView/wallet/persistence/WalletRegistrySpec.scala (hunk 3 above)
Comment (kushti) [reply src/test/scala/org/ergoplatform/nodeView/wallet/persistence/WalletRegistrySpec.scala:216]: Registry is a versioned storage, putBox just puts ke-value pairs into a bag for a versioned update.

**Comment by kushti** [comment]
@aslesarenko all your comments are addressed or answered, please make another pass!

**Review by aslesarenko — APPROVED** [review]

**Pull request merged**